src/actions/project-text-search.js

```diff
@@ -34,6 +36,14 @@ export function clearSearchResults() {
   };
 }
 
+export function closeProjectSearch() {
+  return ({ dispatch, getState }: ThunkArgs) => {
+    dispatch(clearSearchQuery());
```

Comment:
I think I would prefer having the reducers listen for the actions rather than a thunk that fires three.


```js
export function closeProjectSearch() {
  return { type: 'CLOSE_PROJECT_SEARCH' }
}
```

Reply:
In the project search reducer...

```js
    case "CLOSE_PROJECT_SEARCH":
      return state.merge({
        query: "",
        results: state.get("results").clear()
      });
```

in the ui reducer

```js
    case "CLOSE_PROJECT_SEARCH": {
      if (state.get("activeSearch") === "project") {
        return state.set("activeSearch", null);
      }
      return state;
    }
```